### PR TITLE
Task 6: added routing + added changes after previous code review

### PR DESCRIPTION
**Pull request by Libida**
added routing + added changes after previous code review

### CODE REVIEW

File: src/client/actions/movies/index.js [hunk 1]
```diff
@@ -46,14 +39,23 @@ export function updateSortBy(sortBy) {
...
     }
-};
+}
+
+export function resetMoviesSearch() {
+    return function (dispatch) {
+        dispatch(updateSearchTerm(''));
+        dispatch(updateSortBy(SORT_BY_DEFAULT_VALUE));
+        dispatch(updateSearchBy(SEARCH_BY_DEFAULT_VALUE));
+        dispatch(showMovies({data: {data: []}}));
```
Comment (dpryshchep) [inline src/client/actions/movies/index.js:56]: Just advice, all this actions can be applied in scope of one action in the reducer.

File: src/client/components/ButtonGroup/ButtonGroup.jsx [hunk 2]
```diff
@@ -42,3 +45,5 @@ ButtonGroup.defaultProps = {
     incomeClasses: '',
 };
 
+export default React.memo(ButtonGroup, (props1, props2) => (props1.id === props2.id && props1.selectedValue === props2.selectedValue && props1.isDisabled === props2.isDisabled));
```
Comment (dpryshchep) [inline src/client/components/ButtonGroup/ButtonGroup.jsx:48]: please avoid such long one-line conditions

File: src/client/pages/MovieDetailsPage/MovieDetailsPage.jsx [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+
+    return {
+        movie
+    }
+};
+
+const mapDispatchToProps = dispatch => {
+    return {
+        getMovie: (id) => dispatch(getMovie(id)),
+        dispatch
```
Comment (dpryshchep) [inline src/client/pages/MovieDetailsPage/MovieDetailsPage.jsx:52]: i'm not sure that it's necessary declaration

File: src/client/pages/MovieDetailsPage/MovieDetailsPage.jsx [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+        super(props);
+        this.state = {
+            hasError: false
+        };
+    }
+
+    componentDidMount() {
+        const {match} = this.props;
+
+        this.props.getMovie(match.params.id);
```
Comment (dpryshchep) [inline src/client/pages/MovieDetailsPage/MovieDetailsPage.jsx:18]: i would suggest to use lodash `get` function for such case of `id` getting. In such case we will avoid "cannot read property id of undefined, etc."

```
import { get } from 'lodash';

...

const movieId = get(match, 'params.id', 0);
```

File: src/client/pages/MoviesListingPage/MoviesListingPage.jsx [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
+            this.props.resetMoviesSearch();
+        }
+    }
+
+    componentDidUpdate(prevProps, prevState, snapshot) {
+        const {isHome} = this.props;
+
+        // On Home return everything to initial empty state
+        if (isHome && (isHome !== prevProps.isHome)) {
+            console.log("!!!!!!!!RESET");
```
Comment (dpryshchep) [inline src/client/pages/MoviesListingPage/MoviesListingPage.jsx:49]: nope

**Comment by dpryshchep** [comment]
Looks good, task scope completed, just minor advice.

File: src/client/pages/MovieDetailsPage/MovieDetailsPage.jsx [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+    const {movie} = movieDetails;
+
+    return {
+        movie
+    }
+};
+
+const mapDispatchToProps = dispatch => {
+    return {
+        getMovie: (id) => dispatch(getMovie(id)),
```
Comment (dpryshchep) [inline src/client/pages/MovieDetailsPage/MovieDetailsPage.jsx:51]: getMovie,

File: src/client/pages/MoviesListingPage/MoviesListingPage.jsx [hunk 7]
```diff
@@ -0,0 +1,157 @@
...
+            this.props.resetMoviesSearch();
+        }
+    }
+
+    handleSearchTerm(target) {
```
Comment (dpryshchep) [inline src/client/pages/MoviesListingPage/MoviesListingPage.jsx:54]:
```
handleSearchTerm = (target) => {
   this.props.updateSearchTerm(target);
}
```

File: src/client/components/ButtonGroup/ButtonGroup.jsx [hunk 8]
```diff
@@ -11,24 +11,27 @@ function ButtonGroupFunc({id, groupArray, handler, selectedValue, incomeClasses}
         const isChecked = (selectedValue.toLowerCase() === valueLowerCase);
         const idPostfix = `${id}-${index}`;
 
+        console.log(`selectedValue = ${selectedValue}`);
```
Comment (dpryshchep) [inline src/client/components/ButtonGroup/ButtonGroup.jsx:14]: remove

File: src/client/pages/MovieDetailsPage/MovieDetailsPage.jsx (hunk 6 above)
Comment (Libida) [reply src/client/pages/MovieDetailsPage/MovieDetailsPage.jsx:51]: Understood. But will leave this variant, https://react-redux.js.org/using-react-redux/connect-mapdispatch

File: src/client/pages/MovieDetailsPage/MovieDetailsPage.jsx (hunk 3 above)
Comment (Libida) [reply src/client/pages/MovieDetailsPage/MovieDetailsPage.jsx:52]: Understood. But will leave this variant, https://react-redux.js.org/using-react-redux/connect-mapdispatch

**Pull request merged**